ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -1062,29 +1117,62 @@ mcap::SchemaId McapHandler::get_schema_id_nts_(
     }
 }
 
+void McapHandler::rewrite_channels_nts_()
+{
+    logInfo(DDSRECORDER_MCAP_HANDLER, "Rewriting received channels.");
+
+    // Sort channels by id
+    std::map<mcap::ChannelId, mcap::Channel> channels;
+
+    for (auto& channel : channels_)
+    {
+        channels[channel.second.id] = channel.second;
+    }
+
+    // Write channels to MCAP file
+    for (const auto& it : channels)
+    {
+        const auto& id = it.first;
+        const auto& channel = it.second;
+
+        // Check if there is enough space available to write the channel
+        mcap_size_tracker_.channel_to_write(channel);
+
+        // WARNING: passing as non-const to MCAP library
+        mcap_writer_.addChannel(const_cast<mcap::Channel&>(channel));
+        mcap_size_tracker_.channel_written(channel);
+
+        logInfo(DDSRECORDER_MCAP_HANDLER, "Channel created: " << id << ".");
+    }
+}
+
 void McapHandler::rewrite_schemas_nts_()
 {
     logInfo(DDSRECORDER_MCAP_HANDLER, "Rewriting received schemas.");
 
-    std::map<std::string, mcap::Schema> new_schemas;
-    for (const auto& schema : schemas_)
+    // Sort schemas by id
+    std::map<mcap::SchemaId, mcap::Schema> schemas;
+
+    for (auto& schema : schemas_)
     {
-        std::string type_name = schema.first;
-        mcap::Schema new_schema = schema.second;
+        schemas[schema.second.id] = schema.second;
+    }
+
+    // Write schemas to MCAP file
+    for (const auto& it : schemas)
+    {
+        const auto& id = it.first;
+        const auto& schema = it.second;
 
         // Check if there is enough space available to write the schema
-        mcap_size_tracker_.schema_to_write(new_schema);
+        mcap_size_tracker_.schema_to_write(schema);
 
         // WARNING: passing as non-const to MCAP library
-        mcap_writer_.addSchema(new_schema);
-        mcap_size_tracker_.schema_written(new_schema);
-        new_schemas[type_name] = std::move(new_schema);
+        mcap_writer_.addSchema(const_cast<mcap::Schema&>(schema));
+        mcap_size_tracker_.schema_written(schema);
 
-        logInfo(DDSRECORDER_MCAP_HANDLER, "Schema created: " << type_name << ".");
+        logInfo(DDSRECORDER_MCAP_HANDLER, "Schema created: " << id << ".");
```

Comment:
The `type_name` is not trivial to log now since the map doesn't contain it. To log it, I would have to recalculate the map. Would the loss in performance be worth it?

Reply:
Can't you just print `schema.name`? In the case of channels it's true we would be losing the type name information, but it would enough if you print the topic name I would say.